File: apps/web/src/types/project.ts
export type ProjectStatus =
  | "In Progress"
  | "Case Study"
  | "Draft"
  | "Private Repository"
  | "Live Demo"
  | "Coming Soon";

export type ProjectCategory =
  | "Fullstack"
  | "Frontend"
  | "Backend"
  | "Data"
  | "AI Workflow"
  | "Architecture"
  | "Performance"
  | "Portfolio";

export type ProjectSummary = {
  slug: string;
  title: string;
  description: string;
  summary: string;
  category: ProjectCategory[];
  status: ProjectStatus;
  stack: string[];
  featured: boolean;
  repositoryUrl: string | null;
  liveUrl: string | null;
};

export type Project = ProjectSummary & {
  overview: string;
  purpose: string;
  technicalHighlights: string[];
  implementationNotes: string;
  nextSteps: string[];
};

export type ProjectDetail = Project;

File: apps/api/src/Portfolio.Domain/Projects/Project.cs
namespace Portfolio.Domain.Projects;

public sealed class Project
{
    public Project(
        string slug,
        string title,
        string description,
        string summary,
        IReadOnlyCollection<ProjectCategory> category,
        ProjectStatus status,
        IReadOnlyCollection<string> stack,
        bool featured,
        string? repositoryUrl,
        string? liveUrl,
        string overview,
        string purpose,
        IReadOnlyCollection<string> technicalHighlights,
        string implementationNotes,
        IReadOnlyCollection<string> nextSteps)
    {
        Slug = slug;
        Title = title;
        Description = description;
        Summary = summary;
        Category = category;
        Status = status;
        Stack = stack;
        Featured = featured;
        RepositoryUrl = repositoryUrl;
        LiveUrl = liveUrl;
        Overview = overview;
        Purpose = purpose;
        TechnicalHighlights = technicalHighlights;
        ImplementationNotes = implementationNotes;
        NextSteps = nextSteps;
    }

    public string Slug { get; }
    public string Title { get; }
    public string Description { get; }
    public string Summary { get; }
    public IReadOnlyCollection<ProjectCategory> Category { get; }
    public ProjectStatus Status { get; }
    public IReadOnlyCollection<string> Stack { get; }
    public bool Featured { get; }
    public string? RepositoryUrl { get; }
    public string? LiveUrl { get; }
    public string Overview { get; }
    public string Purpose { get; }
    public IReadOnlyCollection<string> TechnicalHighlights { get; }
    public string ImplementationNotes { get; }
    public IReadOnlyCollection<string> NextSteps { get; }
}
